contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -485,7 +492,10 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
         onlyStaker(validatorId)
     {
         address _signer = pubToAddress(signerPubkey);
-        require(_signer != address(0x0) && signerToValidator[_signer] == 0);
+        require(_signer != address(0x0) && signerToValidator[_signer] == 0, "Invalid Signer!");
+        if (latestSignerUpdateEpoch[validatorId] > 0) {
+            require(epoch() >= latestSignerUpdateEpoch[validatorId], "Invalid checkpoint number!");
```

Comment:
for below comment use something like
`require(epoch() >= latestSignerUpdateEpoch[validatorId].add(signerUpdateLimit);, "Invalid checkpoint number!");`

Reply:
it'll check the epoch after adding the limit. In that case, it'll not revert